packages/website/src/components/editor/LoadedEditor.tsx

```diff
@@ -1,387 +1,159 @@
-      }),
-    ];
-
-    return (): void => {
-      for (const subscription of subscriptions) {
-        if (subscription) {
-          subscription.dispose();
+          system.writeFile(activeUri.path, code ?? '');
+        } else {
+          code = system.readFile(activeUri.path);
         }
+        model = monaco.editor.createModel(
+          code ?? '',
+          determineLanguage(activeUri.path),
+          activeUri,
+        );
+        model.updateOptions({ tabSize: 2, insertSpaces: true });
       }
-    };
-  }, [
-    codeActions,
-    main.languages,
-    onChange,
-    onSelect,
-    sandboxInstance.editor,
-    sandboxInstance.monaco.editor,
-    sandboxInstance.monaco.languages.json.jsonDefaults,
-    tabs.code,
-    tabs.eslintrc,
-    tabs.tsconfig,
-    updateMarkers,
-    webLinter.ruleNames,
-  ]);
-
-  const resize = useMemo(() => {
-    return debounce(() => sandboxInstance.editor.layout(), 1);
-  }, [sandboxInstance]);
-
-  const container =
-    sandboxInstance.editor.getContainerDomNode?.() ??
-    sandboxInstance.editor.getDomNode();
-
-  useResizeObserver(container, () => {
-    resize();
-  });
 
-  useEffect(() => {
-    if (
-      !sandboxInstance.editor.hasTextFocus() &&
-      code !== tabs.code.getValue()
-    ) {
-      tabs.code.applyEdits([
-        {
-          range: tabs.code.getFullModelRange(),
-          text: code,
-        },
-      ]);
-    }
-  }, [sandboxInstance, code, tabs.code]);
-
-  useEffect(() => {
-    if (
-      !sandboxInstance.editor.hasTextFocus() &&
-      tsconfig !== tabs.tsconfig.getValue()
-    ) {
-      tabs.tsconfig.applyEdits([
-        {
-          range: tabs.tsconfig.getFullModelRange(),
-          text: tsconfig,
-        },
-      ]);
-    }
-  }, [sandboxInstance, tabs.tsconfig, tsconfig]);
+      setCurrentFile({
+        path: activeFile,
+        language: determineLanguage(activeFile),
+        value: system.readFile(activeUri.path),
+      });
 
-  useEffect(() => {
-    if (
-      !sandboxInstance.editor.hasTextFocus() &&
-      eslintrc !== tabs.eslintrc.getValue()
-    ) {
-      tabs.eslintrc.applyEdits([
-        {
-          range: tabs.eslintrc.getFullModelRange(),
-          text: eslintrc,
-        },
-      ]);
+      editorRef.current.setModel(model);
     }
-  }, [sandboxInstance, eslintrc, tabs.eslintrc]);
 
-  useEffect(() => {
-    sandboxInstance.monaco.editor.setTheme(
-      colorMode === 'dark' ? 'vs-dark' : 'vs-light',
-    );
-  }, [colorMode, sandboxInstance]);
+    monaco.editor.setModelLanguage(model!, determineLanguage(activeUri.path));
+  }, [system, currentFile.path, monaco, editorRef, activeFile]);
+
+  const onEditorDidMount = (
+    editor: Monaco.editor.IStandaloneCodeEditor,
+  ): void => {
+    editorRef.current = editor;
+
+    addLibFiles(system, monaco)
+      .then(() => {
+        window.esquery = utils.esquery;
+        // @ts-expect-error: TODO: remove me, this is only used for debugging
+        window.system = system;
```

Comment:
I kinda like exposing these as globals, as a nice little dev helper for folks... 😄